crates/typed-store/src/rocks/mod.rs

```diff
@@ -373,34 +497,41 @@ impl<K, V> DBMap<K, V> {
             .and_then(|f| f.to_str())
             .unwrap_or("unknown")
             .to_string();
-        let mem_usage_stats = rocksdb::perf::get_memory_usage_stats(Some(&[rocksdb]), None);
-        db_metrics
-            .rocksdb_mem_table_usage
-            .with_label_values(&[&db_name])
-            .set(
-                mem_usage_stats
-                    .as_ref()
-                    .map(|x| x.mem_table_total as i64)
-                    .unwrap_or(METRICS_ERROR),
-            );
-        db_metrics
-            .rocksdb_unflushed_mem_table_usage
-            .with_label_values(&[&db_name])
-            .set(
-                mem_usage_stats
-                    .as_ref()
-                    .map(|x| x.mem_table_unflushed as i64)
-                    .unwrap_or(METRICS_ERROR),
-            );
-        db_metrics
-            .rocksdb_table_readers_usage
-            .with_label_values(&[&db_name])
-            .set(
-                mem_usage_stats
-                    .as_ref()
-                    .map(|x| x.mem_table_readers_total as i64)
-                    .unwrap_or(METRICS_ERROR),
-            );
+
+        if let RocksDB::DBWithThreadMode(ref rocksdb) = **rocksdb {
```

Comment:
Are we sure these metrics are not available for optimistc tx db?

Reply:
we use external library that only supports RocksDB::DBWithThreadMode. We can put TODO to find another way to collect stats for optimistic db